runtime/src/main/java/org/corfudb/runtime/collections/UFOTxnBase.java

```diff
@@ -0,0 +1,369 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.exceptions.StaleRevisionUpdateException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * Thin shim layer around CorfuStore's TxnContext for providing metadata management
+ * capabilities.
+ *
+ * Created by hisundar on 2020-09-16
+ */
+@Slf4j
+class UFOTxnBase implements AutoCloseable {
```

Comment:
why is this here ? should this not be in nsx.

Reply:
if Corfu API changes we want to have test that catch and fail it right away.